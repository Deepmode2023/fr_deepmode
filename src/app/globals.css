@tailwind base;
@tailwind utilities;
@tailwind components;

.min-h-100vh {
  min-height: 100vh;
}
/*
html.dark .bg-global {
  background-color: #221814;
}

html.dark .color-global {
  color: #221814;
}

html.dark .stroke-global {
  stroke: #221814;
}

html.light .bg-global {
  background-color: white;
  stroke: black;
} */
