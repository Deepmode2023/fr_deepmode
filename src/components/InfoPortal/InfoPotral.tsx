import React from "react";

import { IInfoPortalProps } from "@/interfaces/components/infoportal";
import ClientPortal from "../Portal/Portal";
import { PORTAL_ID } from "@/global.constant";
import { motion } from "framer-motion";

export const InfoPortal: React.FC<IInfoPortalProps> = ({
  children,
  isActiveLayout,
}) => {
  return (
    <ClientPortal
      selector={PORTAL_ID.LAYOUT_PORTAL.INFO_CONTENT}
      show={isActiveLayout}
    >
      <div
        data-testid="info-portal-test-id"
        className="absolute top-0 bottom-0 right-0 left-0 z-50"
      >
        <div className="sticky top-[110px] h-[calc(100vh-110px)] flex items-center justify-center">
          <motion.div
            className="bg-red-300"
            initial={{ width: 0, height: 0 }}
            animate={{ width: "300px", height: 20 }}
            transition={{
              duration: 0.5,
              delay: 0.2,
            }}
          >
            {children}
          </motion.div>
        </div>
      </div>
    </ClientPortal>
  );
};
