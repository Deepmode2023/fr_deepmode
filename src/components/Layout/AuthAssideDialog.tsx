"use client";
import React from "react";
import { cls } from "@/shared";
import { LeftMultiplySvg } from "@/assets/icons/navigate";

import { AuthAssideStore } from "@/zustand/authAssideStore";
import { createSelectorHooks } from "auto-zustand-selectors-hook";
import { NavbarStore } from "@/zustand/navbarStore";

const navbarStore = createSelectorHooks(NavbarStore);
const authAssideStore = createSelectorHooks(AuthAssideStore);

const AuthAssideDialog = () => {
  const isActive = authAssideStore.useIsActive();
  const assideContent = authAssideStore.useAssideContent();
  const assideHeader = authAssideStore.useAssideHeader();
  const changeAssideDialog = authAssideStore.useChangeAssideDialog();
  const changeAuthActiveLink = navbarStore.useChangeAuthActiveLink();

  return (
    <div
      className={cls(
        "w-[500px] h-[700px] rounded-xl z-10 absolute p-[20px] pt-0 flex flex-col gap-[10px] overflow-hidden overflow-y-auto scroll-left light-scroll-color",
        "dark:bg-dark-total bg-light-total transition-transform duration-700",
        isActive ? "translate-x-[-410px]" : "translate-x-[410px]"
      )}
    >
      <div
        className={cls(
          "flex gap-3 justify-between items-center border-b-2 pb-[5px] pt-[20px] z-10",
          "dark:border-dark-color3 border-light-color1",
          "sticky top-0 dark:bg-dark-total bg-light-total"
        )}
      >
        {assideHeader ?? <div></div>}
        <div
          className={cls(
            "right-[20px] top-[20px] cursor-pointer rounded-md p-[5px]",
            "rotate-180 bg-light-color1 dark:bg-dark-color3",
            isActive && "rotate-0"
          )}
          onClick={() => {
            changeAuthActiveLink("main");
            changeAssideDialog(null, null, false);
          }}
        >
          <LeftMultiplySvg strokeColor="stroke-white translate-x-[-5px]" />
        </div>
      </div>
      <div className={cls(isActive ? "opacity-100" : "opacity-0", "z-0")}>
        {assideContent}
      </div>
    </div>
  );
};

export { AuthAssideDialog };
