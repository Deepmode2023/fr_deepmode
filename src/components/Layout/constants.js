export const SIZED_LAYOUT = {
  MAIN: { gap: 4 },
  NAVBAR: { minWidth: 100, maxWidth: 300 },
  AUTH_NAVBAR: { height: 100 },
};

export const LAYOUT_GRID = {
  DESC: {
    main: `grid-cols-[auto_1fr_auto_100px] grid-rows-[${
      SIZED_LAYOUT.AUTH_NAVBAR.height / 2
    }px_${
      SIZED_LAYOUT.AUTH_NAVBAR.height / 2
    }px_minmax(100px,300px)_auto] box-border gap-1`,
    // NAVBAR
    navbar: `col-start-1 col-end-1 row-start-1 row-end-5 min-w-[${SIZED_LAYOUT.NAVBAR.minWidth}px] 
                max-w-[${SIZED_LAYOUT.NAVBAR.maxWidth}px] relative box-border`,
    // AUTH NAVBAR
    authNavbar:
      "col-start-3 col-end-5  row-start-1 row-end-3 sticky top-0 z-50 relative box-border",
    // CONTENT
    content:
      "col-start-2 col-end-5 row-start-3 row-end-5 z-0 box-border relative",
    // NAVBAR TOOLTIP
    navbarTooltip:
      "col-start-2 col-end-5 row-start-3 row-end-4 z-60 box-border sticky top-0",
  },
};
