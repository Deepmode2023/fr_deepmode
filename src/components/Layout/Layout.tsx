import React from "react";
import { mergeCls } from "@/utils/cls";
import { PORTAL_ID } from "@/global.constant";
import { LAYOUT_GRID } from "./constants";

import { AuthAssideDialog } from "./AuthAssideDialog";

export interface ILayoutProps {
  height?: number;
  isDev?: boolean;
  children: React.ReactNode;
  Navbar: React.ReactNode;
  AuthNavbarAsside: React.ReactNode;
  AuthNavbar: React.ReactNode;
}

export const Layout: React.FC<ILayoutProps> = ({
  height,
  isDev = true,
  children,
  AuthNavbar,
  AuthNavbarAsside,
  Navbar,
}) => {
  const dialogContent = "";
  return (
    <main
      style={{ height: height ?? "100%" }}
      className={mergeCls("w-full grid", LAYOUT_GRID.DESC.main)}
    >
      <aside
        id={PORTAL_ID.LAYOUT_PORTAL.NAVBAR}
        className={mergeCls(
          LAYOUT_GRID.DESC.navbar,
          isDev && "border-2 border-orange-700"
        )}
      >
        {Navbar}
      </aside>
      <article
        id={PORTAL_ID.LAYOUT_PORTAL.AUTH_NAVBAR}
        className={mergeCls(
          LAYOUT_GRID.DESC.authNavbar,
          isDev && "border-2 border-red-600"
        )}
      >
        {AuthNavbar}
      </article>
      <aside className={mergeCls(LAYOUT_GRID.DESC.authNavbarAsside, "z-20")}>
        <AuthAssideDialog />
        <div className="sticky top-[105px] group-auth-asside">
          {AuthNavbarAsside}
        </div>
      </aside>
      <article
        id={PORTAL_ID.LAYOUT_PORTAL.INFO_CONTENT}
        className={mergeCls(
          LAYOUT_GRID.DESC.content,
          isDev && "border-2 border-green-400"
        )}
      >
        <div className="z-0">{children}</div>
      </article>
    </main>
  );
};
