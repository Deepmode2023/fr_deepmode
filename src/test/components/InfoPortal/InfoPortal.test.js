import { expect } from "@jest/globals";
import React from "react";
import { InfoPortal } from "@/components/InfoPortal/InfoPotral";
import { Layout } from "@/components/Layout/Layout";
import { render, waitFor, queryByTestId } from "@testing-library/react";

const AuthNavbar = () => <div>AuthNavbar</div>;
const Navbar = () => <div>Navbar</div>;

const propsInfo = { isActiveLayout: false };
const propsLayout = { AuthNavbar: <AuthNavbar />, Navbar: <Navbar /> };

test("Checking props 'isActiveLayout' with false value", () => {
  const componentInfo = render(
    <InfoPortal {...propsInfo}>
      <div>Harry</div>
    </InfoPortal>
  );
  const componentLayout = render(
    <Layout {...propsLayout}>
      <div>Button</div>
    </Layout>
  );

  expect(
    queryByTestId(componentLayout.container, "info-portal-test-id")
  ).not.toBeInTheDocument();
});

test("Checking props 'isActiveLayout' with true value and displayed portal inside Layout", async () => {
  const componentLayout = render(
    <Layout {...propsLayout}>
      <div>Button</div>
    </Layout>
  );

  const { rerender } = render(
    <InfoPortal {...propsInfo}>
      <div>Harry</div>
    </InfoPortal>
  );

  expect(
    queryByTestId(componentLayout.container, "info-portal-test-id")
  ).not.toBeInTheDocument();

  waitFor(() => {
    rerender(
      <InfoPortal isActiveLayout={true}>
        <div>Harry</div>
      </InfoPortal>
    );
  });

  expect(
    queryByTestId(componentLayout.container, "info-portal-test-id")
  ).toBeInTheDocument();

  expect(componentLayout).toMatchSnapshot();
});
