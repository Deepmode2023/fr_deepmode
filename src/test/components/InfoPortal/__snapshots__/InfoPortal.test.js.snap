// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking props 'isActiveLayout' with true value and displayed portal inside Layout 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class=" w-full grid grid-cols-[auto_1fr_auto_100px] grid-rows-[50px_50px_minmax(100px,300px)_auto] box-border gap-1"
        style="height: 100%;"
      >
        <aside
          class="col-start-1 col-end-1 row-start-1 row-end-5 min-w-[100px] 
                max-w-[300px] relative box-border border-2 border-orange-700"
          id="navbar_portal_id"
        >
          <div>
            Navbar
          </div>
        </aside>
        <article
          class="col-start-3 col-end-5  row-start-1 row-end-3 sticky top-0 z-50 relative box-border border-2 border-red-600"
          id="auth_navbar_portal_id"
        >
          <div>
            AuthNavbar
          </div>
        </article>
        <article
          class="col-start-2 col-end-5 row-start-3 row-end-5 z-0 box-border relative border-2 border-green-400"
          id="info_content_portal_id"
        >
          <div
            class="z-0"
          >
            <div>
              Button
            </div>
          </div>
          <div
            class="absolute top-0 bottom-0 right-0 left-0 z-50"
            data-testid="info-portal-test-id"
          >
            <div
              class="sticky top-[110px] h-[calc(100vh-110px)] flex items-center justify-center"
            >
              <div
                class="bg-red-300"
                style="width: 0px; height: 0px;"
              >
                <div>
                  Harry
                </div>
              </div>
            </div>
          </div>
        </article>
      </main>
    </div>
    <div />
  </body>,
  "container": <div>
    <main
      class=" w-full grid grid-cols-[auto_1fr_auto_100px] grid-rows-[50px_50px_minmax(100px,300px)_auto] box-border gap-1"
      style="height: 100%;"
    >
      <aside
        class="col-start-1 col-end-1 row-start-1 row-end-5 min-w-[100px] 
                max-w-[300px] relative box-border border-2 border-orange-700"
        id="navbar_portal_id"
      >
        <div>
          Navbar
        </div>
      </aside>
      <article
        class="col-start-3 col-end-5  row-start-1 row-end-3 sticky top-0 z-50 relative box-border border-2 border-red-600"
        id="auth_navbar_portal_id"
      >
        <div>
          AuthNavbar
        </div>
      </article>
      <article
        class="col-start-2 col-end-5 row-start-3 row-end-5 z-0 box-border relative border-2 border-green-400"
        id="info_content_portal_id"
      >
        <div
          class="z-0"
        >
          <div>
            Button
          </div>
        </div>
        <div
          class="absolute top-0 bottom-0 right-0 left-0 z-50"
          data-testid="info-portal-test-id"
        >
          <div
            class="sticky top-[110px] h-[calc(100vh-110px)] flex items-center justify-center"
          >
            <div
              class="bg-red-300"
              style="width: 0px; height: 0px;"
            >
              <div>
                Harry
              </div>
            </div>
          </div>
        </div>
      </article>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
